tests/utils/test_geometry_comparisons.py

```diff
@@ -3,55 +3,207 @@
+POINT3_RELATIVE_ERROR_THRESH = 1e-1
+POINT3_ABS_ERROR_THRESH = 1e-2
+
+
+def rot3_compare(R: Rot3, R_: Rot3, msg=None) -> bool:
+    return np.abs(geometry_comparisons.compute_relative_rotation_angle(R, R_)) < ROT3_EQUALITY_THRESH
+
+
+def point3_compare(t: Point3, t_: Point3, msg=None) -> bool:
+    return np.allclose(t, t_, rtol=POINT3_RELATIVE_ERROR_THRESH, atol=POINT3_ABS_ERROR_THRESH)
+
 
 class TestGeometryComparisons(unittest.TestCase):
     """Unit tests for comparison functions for geometry types."""
 
-    def test_compare_poses_exact(self):
-        """Check pose comparison with exactly same inputs."""
-        self.assertTrue(geometry_comparisons.compare_global_poses(POSE_LIST, POSE_LIST))
+    def __assert_equality_on_rot3s(self, computed: List[Rot3], expected: List[Rot3]) -> None:
+
+        self.assertEqual(len(computed), len(expected))
+
+        for R, R_ in zip(computed, expected):
+            self.assertEqual(R, R_)
 
-    def test_compare_poses_with_uniform_scaled_translations(self):
-        """Check pose comparison with all translations in input #2 scaled by
-        the same scalar factor."""
-        scale_factor = 1.2
-        pose_list_ = [Pose3(x.rotation(), x.translation() * scale_factor) for x in POSE_LIST]
+    def __assert_equality_on_point3s(self, computed: List[Point3], expected: List[Point3]) -> None:
 
-        self.assertTrue(geometry_comparisons.compare_global_poses(POSE_LIST, pose_list_))
+        self.assertEqual(len(computed), len(expected))
 
-    def test_compare_poses_with_nonuniform_scaled_translations(self):
-        """Check pose comparison with all translations in input #2 scaled by
-        significantly different scalar factors."""
-        scale_factors = [0.3, 0.7, 0.9, 1.0, 1.0, 0.99, 1.01, 1.10]
-        pose_list_ = [Pose3(x.rotation(), x.translation() * scale_factors[idx]) for idx, x in enumerate(POSE_LIST)]
+        for t, t_ in zip(computed, expected):
+            np.testing.assert_allclose(t, t_, rtol=POINT3_RELATIVE_ERROR_THRESH, atol=POINT3_ABS_ERROR_THRESH)
 
-        self.assertFalse(geometry_comparisons.compare_global_poses(POSE_LIST, pose_list_))
+    def __assert_equality_on_pose3s(self, computed: List[Pose3], expected: List[Pose3]) -> None:
 
-    def test_compare_poses_with_origin_shift(self):
-        """Check pose comparison with a shift in the global origin."""
-        new_origin = Pose3(Rot3.RzRyRx(0.3, 0.1, -0.27), np.array([-20.0, +19.0, 3.5]))
+        self.assertEqual(len(computed), len(expected))
 
-        pose_list_ = [new_origin.between(x) for x in POSE_LIST]
+        computed_rot3s = [x.rotation() for x in computed]
+        computed_point3s = [x.translation() for x in computed]
+        expected_rot3s = [x.rotation() for x in expected]
+        expected_point3s = [x.translation() for x in expected]
 
-        self.assertTrue(geometry_comparisons.compare_global_poses(POSE_LIST, pose_list_))
+        self.__assert_equality_on_rot3s(computed_rot3s, expected_rot3s)
```

Comment:
whats the advantage of splitting these up, rather than just using `compare_global_poses()`? Do you mind noting that in a comment?

Reply:
There is no advantage. Here I am not performing any alignment and just comparing that values should be same. I cannot use `compare_global_poses` because it wont serve the purpose of unit tests. 

Infact I will revert from using the angle of relative rotations to the Euler angles.